components/level/game.tsx

```diff
@@ -447,6 +447,11 @@ export default function Game({
 
       if (newGameState.board[newGameState.pos.y][newGameState.pos.x].tileType === TileType.End) {
         // track stats upon reaching an exit
+
+        if (gameState.moves.length <= level.leastMoves && onComplete) {
+          onComplete();
```

Comment:
moving this here fixes an issue where on refresh if you have solved already it re-fires the onComplete